pkg/storage/watch_cache_test.go

```diff
@@ -230,6 +231,25 @@ func TestEvents(t *testing.T) {
 	}
 }
 
+func TestWaitUntilFreshAndList(t *testing.T) {
+	store := newWatchCache(3)
+
+	// In background, wait some time and then update the store.
```

Comment:
If that's what you want to test, we should think of a way to always test that... not probably. You'd be surprised how very very high probability doesn't always happen, especially when it's time based in a resource constrained CI system running thousands of tests.